lib/Doctrine/ODM/MongoDB/Utility/EncryptedFieldsMapGenerator.php

```diff
@@ -49,7 +86,7 @@ private function createEncryptionFieldMap(ClassMetadata $classMetadata, string $
                 if ($embedMetadata->isEncrypted) {
                     $mapping['encrypt'] ??= []; // @todo get the keyId
                 } elseif (! isset($mapping['encrypt'])) {
-                    yield from $this->createEncryptionFieldMap(
+                    yield from $this->createEncryptedFieldsMapForClass(
                         $embedMetadata,
                         $path . $mapping['name'] . '.',
                     );
```

Comment:
Instead of the class property `$generatorStack`, I would use a method parameter so that the scope is clear and you don't have to unset the class once it's handled.

```suggestion
                    yield from $this->createEncryptedFieldsMapForClass(
                        $embedMetadata,
                        $path . $mapping['name'] . '.',
                        $visitedClasses + [$classMetadata->getName() => true],
                    );
```

As an improvement: using an `int` level counter instead of `true`, we can set a limit in the recursion deep (if necessary).

Reply:
Changed the recursion prevention to a method parameter. Not sure how many recursion levels would be ok, but for the time being I'll prevent any recursion.